mitodl/odl-video-service
# Add USwitch player to OVS

#### What are the relevant tickets?
Fixes #93

#### What's this PR do?
- Adds a boolean field to the Video model ('multiangle')
- if the video is multiangle, loads the USwitch player instead of the video.js player

#### How should this be manually tested?
- Add this line to your `.env` file: `USWITCH_URL=https://s3.amazonaws.com/odl-video-service-uswitch-dev`
- `docker-compose build`
- `docker-compose run`
- Using the admin console, set 'multiangle=True' on any video
- Modify your `/etc/hosts` so that one of the following domains (`localhost`, `mit.edu`, `amps-web.amps.ms.mit.edu`) points to your docker host, for example `192.168.99.100 amps-web.amps.ms.mit.edu`
- Load the video detail page for the modified video, using `amps-web.amps.ms.mit.edu` as the domain.  You should see the USwitch player instead of the video.js player.
- Play the video, there will be one main window and 4 smaller windows at the top showing the 4 corners of the video, labeled Camera 1-4.

#### Any background context you want to provide?
There is an issue with styling on the detail page when the USwitch player is used - #104 
The player only works with certain domains: #108
This assumes the USwitch JS/CSS will be hosted on S3: #103
